src/event-stream/reader.ts

```diff
@@ -182,110 +186,135 @@ export function createTransactionFromCoreBtcTxEvent(
+            coreTx
+          )}`
+        );
+        throw new Error('Unable to generate transaction from BTC tx');
       }
-      const parsedTx: CoreNodeParsedTxMessage = {
-        core_tx: coreTx,
-        nonce: Number(rawTx.auth.originCondition.nonce),
-        raw_tx: txBuffer,
-        parsed_tx: rawTx,
-        block_hash: msg.block_hash,
-        index_block_hash: msg.index_block_hash,
-        block_height: msg.block_height,
-        burn_block_time: msg.burn_block_time,
-        sender_address: txSender,
-        sponsor_address: sponsorAddress,
-      };
-      parsedMessage.parsed_transactions[i] = parsedTx;
-      const payload = rawTx.payload;
-      switch (payload.typeId) {
-        case TransactionPayloadTypeID.Coinbase: {
-          break;
-        }
-        case TransactionPayloadTypeID.SmartContract: {
-          logger.verbose(`Smart contract deployed: ${parsedTx.sender_address}.${payload.name}`);
-          break;
-        }
-        case TransactionPayloadTypeID.ContractCall: {
-          const address = c32address(
-            payload.address.version,
-            payload.address.bytes.toString('hex')
-          );
-          logger.verbose(
-            `Contract call: ${address}.${payload.contractName}.${payload.functionName}`
-          );
-          break;
-        }
-        case TransactionPayloadTypeID.TokenTransfer: {
-          let recipientPrincipal = c32address(
-            payload.recipient.address.version,
-            payload.recipient.address.bytes.toString('hex')
-          );
-          if (payload.recipient.typeId === RecipientPrincipalTypeId.Contract) {
-            recipientPrincipal += '.' + payload.recipient.contractName;
-          }
-          logger.verbose(
-            `Token transfer: ${payload.amount} from ${parsedTx.sender_address} to ${recipientPrincipal}`
-          );
-          break;
-        }
-        default: {
-          throw new NotImplementedError(
-            `extracting data for tx type: ${getEnumDescription(
-              TransactionPayloadTypeID,
-              rawTx.payload.typeId
-            )}`
-          );
+    } else {
+      const bufferReader = BufferReader.fromBuffer(txBuffer);
+      rawTx = readTransaction(bufferReader);
+      txSender = getTxSenderAddress(rawTx);
+      sponsorAddress = getTxSponsorAddress(rawTx);
+    }
+    const parsedTx: CoreNodeParsedTxMessage = {
+      core_tx: coreTx,
+      nonce: Number(rawTx.auth.originCondition.nonce),
+      raw_tx: txBuffer,
+      parsed_tx: rawTx,
+      block_hash: blockData.block_hash,
+      index_block_hash: blockData.index_block_hash,
+      parent_index_block_hash: blockData.parent_index_block_hash,
+      parent_block_hash: blockData.parent_block_hash,
+      block_height: blockData.block_height,
+      burn_block_time: blockData.burn_block_time,
+      microblock_sequence: coreTx.microblock_sequence ?? I32_MAX,
+      microblock_hash: coreTx.microblock_hash ?? '',
```

Comment:
Might it be better to use `null` here?

Reply:
It would definitely be better, but that will probably have to be a new PR. Each small type or property change in a common object like this requires hundreds of lines of code changes, and a ton of time fixing unit/integration tests 🙁